src/compiler/types/lines_500_1000.rs

```diff
@@ -173,7 +174,15 @@ pub trait NodeInterface: ReadonlyTextRange {
     fn maybe_js_doc(&self) -> Option<Vec<Rc<Node /*JSDoc*/>>>;
     fn set_js_doc(&self, js_doc: Vec<Rc<Node /*JSDoc*/>>);
     fn maybe_js_doc_cache(&self) -> Option<Vec<Rc<Node /*JSDocTag*/>>>;
-    fn set_js_doc_cache(&self, js_doc_cache: Vec<Rc<Node /*JSDocTag*/>>);
+    fn set_js_doc_cache(&self, js_doc_cache: Option<Vec<Rc<Node /*JSDocTag*/>>>);
+    // IncrementalElement
+    fn maybe_intersects_change(&self) -> Option<bool>;
```

Comment:
Expose the optional `IncrementalElement`/`IncrementalNode` fields on `NodeInterface`